src/main/java/org/broadinstitute/dsm/route/DisplaySettingsRoute.java

```diff
@@ -65,14 +67,16 @@ public Object processRequest(Request request, Response response, String userId)
             if (StringUtils.isNotBlank(realm) && instance != null && StringUtils.isNotBlank(userIdRequest)
                     && StringUtils.isNotBlank(parent) && StringUtils.isNotBlank(ddpGroupId)) {
                 Map<String, Object> displaySettings = new HashMap<>();
+                InstanceSettings instanceSettings = new InstanceSettings();
                 displaySettings.put("assignees", Assignee.getAssignees(realm));
                 displaySettings.put("fieldSettings", FieldSettings.getFieldSettings(realm));
                 displaySettings.put("drugs", Drug.getDrugList());
                 displaySettings.put("cancers", Cancer.getCancers());
                 displaySettings.put("activityDefinitions", ElasticSearchUtil.getActivityDefinitions(instance));
                 displaySettings.put("filters", ViewFilter.getAllFilters(userIdRequest, patchUtil.getColumnNameMap(), parent, ddpGroupId, instance.getDdpInstanceId()));
                 displaySettings.put("abstractionFields", AbstractionUtil.getFormControls(realm));
-                InstanceSettings instanceSettings = new InstanceSettings();
+                displaySettings.put("hasAddressTab", instanceSettings.getHasAddressTabByStudyInstanceName(realm));
```

Comment:
I don't think so that it is necessary to manually put `hasAddressTab` and `hasComputedObject` to the map, `InstanceSettings` has method already called `getInstanceSettings`(Line 80) which returns all the settings by instance name(realm), then method  `getInstanceSettingsAsMap` of `InstanceSettings` will automatically extract all necessary settings from `instanceSettingsDto` with help of reflection. You would have only to change method `getInstanceSettingsByInstanceName` of `InstanceSettingsDao` by extracting two new table column from `ResultSet` .
InstanceSettingsDao, L133